test/upgrade/prober/verify.go

```diff
@@ -53,26 +55,132 @@ func (p *prober) Finish(ctx context.Context) {
 	p.removeSender(ctx)
 }
 
-func fetchReceiverReport(address *corev1.NodeAddress, log *zap.SugaredLogger) *Report {
-	u := fmt.Sprintf("http://%s:%d/report", address.Address, receiverNodePort)
-	log.Infof("Fetching receiver report from: %v", u)
-	resp, err := http.Get(u)
+func (p *prober) fetchReport(ctx context.Context) *receiver.Report {
+	exec := p.fetchExecution(ctx)
+	replayLogs(p.log, exec)
+	return exec.Report
+}
+
+func replayLogs(log *zap.SugaredLogger, exec *fetcher.Execution) {
+	for _, entry := range exec.Logs {
+		logFunc := log.Error
+		switch entry.Level {
+		case "debug":
+			logFunc = log.Debug
+		case "info":
+			logFunc = log.Info
+		case "warning":
+			logFunc = log.Warn
+		}
+		logFunc("Fetcher: ", entry.Datetime, " ", entry.Message)
+	}
+}
+
+func (p *prober) fetchExecution(ctx context.Context) *fetcher.Execution {
+	ns := p.config.Namespace
+	p.deployFetcher(ctx)
+	defer p.deleteFetcher(ctx)
+	pod := p.succeededJobPod(ctx, fetcherName, ns)
```

Comment:
I wonder if there is a way to get the pod name from `deployFetcher`, the reason being that if somehow the previous job is not cleanup, eg this is run twice without deleting the cluster, this can lead to surprise.

Reply:
I think the pod name should be discovered as well. Good catch

Reply:
Pod names are unique. I'm looking at the first pod that succeeded. Pods might fail to sprung for whatever reason.

Reply:
Pod names are unique, and fetcher job is deployed in unique namespace, and all tests resources are automatically removed after the test. No risk there.